go/v1beta1/storage/elasticsearch.go

```diff
@@ -367,7 +370,40 @@ func (es *ElasticsearchStorage) BatchCreateOccurrences(ctx context.Context, proj
 
 // UpdateOccurrence updates the existing occurrence with the given projectId and occurrenceId
 func (es *ElasticsearchStorage) UpdateOccurrence(ctx context.Context, projectId, occurrenceId string, o *pb.Occurrence, mask *fieldmaskpb.FieldMask) (*pb.Occurrence, error) {
-	return nil, nil
+	occurrenceName := fmt.Sprintf("projects/%s/occurrences/%s", projectId, occurrenceId)
+	log := es.logger.Named("Update Occurrence").With(zap.String("occurrence", occurrenceName))
+
+	search := &esSearch{
+		Query: &filtering.Query{
+			Term: &filtering.Term{
+				"name": occurrenceName,
+			},
+		},
+	}
+
+	occurrence := &pb.Occurrence{}
+
+	targetDocumentID, err := es.genericGet(ctx, log, search, occurrencesIndex(projectId), occurrence)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if o.UpdateTime == nil {
```

Comment:
not sure how I feel about this, but I'm leaning towards always updating the timestamp ourselves whether or not the user provided an `UpdateTime`.  thoughts?

Reply:
I changed my mind - I'd prefer to keep what you already have here